server/service/validation_setup.go

```diff
@@ -27,14 +27,32 @@ func (mw validationMiddleware) NewAppConfig(ctx context.Context, payload fleet.A
 	return mw.Service.NewAppConfig(ctx, payload)
 }
 
-func validateServerURL(urlString string) error {
-	serverURL, err := url.Parse(urlString)
+func ValidateServerURL(urlString string) error {
+	// TODO - implement more robust URL validation here
+
+	// no valid scheme provided
+	if !(strings.HasPrefix(urlString, "http://") || strings.HasPrefix(urlString, "https://")) {
+		parsed, err := url.Parse("https://" + urlString)
+		if err != nil {
+			return err
+		}
+		// localhost only acceptable host if no schem (protocol) provided. .Host will include port (e.g.
+		// 8080), so we check for "localhost" prefix
+		// by checking for prefix, we also invalidate unsupported schemes like "ftp://"
+		if !strings.HasPrefix(parsed.Host, "localhost") {
+			return errors.New(fleet.InvalidServerURLMsg)
+		}
```

Comment:
We can use `parsed.HostName()` to get the host without the port, to ensure it's exactly `localhost`